Add host Ksmtuned to support KSM (#195)
harvester/docs: docs/host/host.md

![Edit Config](/img/v1.1/host/edit-config.png)
![Add Disks](/img/v1.1/host/add-disks.png)

> ## Ksmtuned Mode

Reviewer: ```suggestion
## Ksmtuned Mode

_Available as of v1.1.0_

```

> Ksmtuned is a KSM automation tool deployed as a DaemonSet to run Ksmtuned per node and start or stop KSM by setting the available memory percentage ratio (**Threshold Coefficient**), By default, you need to manually enable Ksmtuned on each node UI. You can see the KSM statistics from the node UI ([see the KSM in detail](https://www.kernel.org/doc/html/latest/admin-guide/mm/ksm.html#ksm-daemon-sysfs-interface))

Reviewer 2: ```suggestion
Ksmtuned is a KSM automation tool deployed as a DaemonSet to run Ksmtuned on each node. It can start or stop KSM by setting the available memory percentage ratio (**Threshold Coefficient**), By default, you need to manually enable Ksmtuned on each node UI. You can see the KSM statistics from the node UI (for details, see [KSM](https://www.kernel.org/doc/html/latest/admin-guide/mm/ksm.html#ksm-daemon-sysfs-interface)).
```

> Ksmtuned is a KSM automation tool deployed as a DaemonSet to run Ksmtuned on each node. It can start or stop KSM by setting the available memory percentage ratio (**Threshold Coefficient**), By default, you need to manually enable Ksmtuned on each node UI. You can see the KSM statistics from the node UI (for details, see [KSM](https://www.kernel.org/doc/html/latest/admin-guide/mm/ksm.html#ksm-daemon-sysfs-interface)).

Reviewer 2: ```suggestion
Ksmtuned is a KSM automation tool deployed as a DaemonSet to run Ksmtuned on each node. It can start or stop KSM by setting the available memory percentage ratio (**Threshold Coefficient**). By default, you need to manually enable Ksmtuned on each node UI. You can see the KSM statistics from the node UI (for details, see [KSM](https://www.kernel.org/doc/html/latest/admin-guide/mm/ksm.html#ksm-daemon-sysfs-interface)).
```

> Ksmtuned is a KSM automation tool deployed as a DaemonSet to run Ksmtuned on each node. It can start or stop KSM by setting the available memory percentage ratio (**Threshold Coefficient**). By default, you need to manually enable Ksmtuned on each node UI. You can see the KSM statistics from the node UI (for details, see [KSM](https://www.kernel.org/doc/html/latest/admin-guide/mm/ksm.html#ksm-daemon-sysfs-interface)).

Reviewer: ```suggestion
Ksmtuned is a KSM automation tool deployed as a DaemonSet to run Ksmtuned on each node. It will start or stop the KSM by watching the available memory percentage ratio (**i.e. Threshold Coefficient**). By default, you need to manually enable Ksmtuned on each node UI. You will be able to see the KSM statistics from the node UI after 1-2 minutes.(check [KSM](https://www.kernel.org/doc/html/latest/admin-guide/mm/ksm.html#ksm-daemon-sysfs-interface) for more details).
```

### **Quick Run**

1. Go to the **Hosts** page.
2. On the node you want to modify, click **⋮ > Edit Config**.
3. Select the **Ksmtuned** tab and **Run Strategy** in Run.

> 4. (Optional)You can change **Threshold Coefficient** to match your expectations.

Reviewer 2: ```suggestion
4. (Optional) You can modify **Threshold Coefficient** as needed.
```

> ![Edit Ksmtuned](/img/v1.1/host/edit-ksmtuned.png)

Reviewer: please update the related image.

5. Click **Save** to update.

> 6. Wait at least 2 minutes and check the **Statistics**, click **⋮ >  Your Node > Ksmtuned tab.**

Reviewer: ```suggestion
6. Wait for about 1-2 minutes and you can check its **Statistics** via click **⋮ >  Your Node > Ksmtuned tab.**
```

Reviewer 2: `via` should be `by clicking`?

![View Ksmtuned Statistics](/img/v1.1/host/view-ksmtuned-statistics.png)

### **Parameters**

**Run Strategy：**

> - **Stop:** Stop Ksmtuned and KSM，VM can still use shared memory pages.

Reviewer 2: ```suggestion
- **Stop:** Stop Ksmtuned and KSM, and VM can still use shared memory pages.
```

> - **Stop:** Stop Ksmtuned and KSM，VM can still use shared memory pages.

Reviewer 2: ```suggestion
- **Stop:** Stop Ksmtuned and KSM. VMs can still use shared memory pages.
```

> - **Run:** Run Ksmtuned。

Reviewer 2: ```suggestion
- **Run:** Run Ksmtuned.
```

- **Prune:** Stop Ksmtuned and prune KSM memory pages.

> **Threshold Coefficient**:  configures the available memory percentage ratio, if the available memory is less than the threshold starts KSM, otherwise, it stops KSM.

Reviewer 2: ```suggestion
**Threshold Coefficient**: configures the available memory percentage ratio. If the available memory is less than the threshold, KSM will be started; otherwise, KSM will be stopped.
```

Reviewer 2: If KSM is started/stopped automatically, then I think using passive voice is better

> **Merge Across Nodes:**  specifies if pages from different NUMA nodes can be merged.

Reviewer 2: ```suggestion
**Merge Across Nodes:** specifies if pages from different NUMA nodes can be merged.
```

**Mode:**

> - **Standard:** For the default mode, the control node ksmd uses a single CPU to about 20%, with the following parameters:

Reviewer 2: ```suggestion
- **Standard:** For the default mode. The control node ksmd uses about 20% of a single CPU. It uses the following parameters:
```

> - **Standard:** For the default mode. The control node ksmd uses about 20% of a single CPU. It uses the following parameters:

Reviewer 2: ```suggestion
- **Standard:** The default mode. The control node ksmd uses about 20% of a single CPU. It uses the following parameters:
```

```yaml
Boost: 0
Decay: 0
Maximum Pages: 100
Minimum Pages: 100
Sleep Time: 20
```

> - **High-performance:** Node ksmd uses from 20% to 100% of the single CPU and has faster scanning and merging efficiency, with the following parameters:

Reviewer 2: ```suggestion
- **High-performance:** Node ksmd uses 20% to 100% of a single CPU and has higher scanning and merging efficiency. It uses the following parameters:
```

```yaml
Boost: 200
Decay: 50
Maximum Pages: 100
Minimum Pages: 10000
Sleep Time: 20
```

- **Customized:** You can customize the configuration to reach the performance that you want.

> Ksmtuned uses the following parameters to control KSM efficiency by configuring:

Reviewer 2: ```suggestion
Ksmtuned uses the following parameters to control KSM efficiency:
```

> - **Boost**: The number of scanned pages is incremented each time, and if the available memory is less than the **Threshold Coefficient**.
> - **Decay**: The number of scanned pages is decremented each time, and if the available memory is greater than the **Threshold Coefficient**.

Reviewer 2: ```suggestion
- **Boost**: The number of scanned pages is incremented each time if the available memory is less than the **Threshold Coefficient**.
- **Decay**: The number of scanned pages is decremented each time if the available memory is greater than the **Threshold Coefficient**.
```

- **Maximum Pages**: Maximum number of pages per scan.
- **Minimum Pages**: The minimum number of pages per scan, also the configuration for the first run.

> - **Sleep Time(ms)**: The interval between two scans, which is calculated with the formula: **Sleep Time** \* 16 \* 1024 \* 1024 / Total Memory. Minimum: 10ms.

Reviewer: - a table view might be preferred here,
- would be better to use a bracket where applicable, e.g., `(Sleep Time * 16 * 1024 * 1024 / Total Memory)`

> - **Sleep Time(ms)**: The interval between each detection, is calculated by the formula (**Sleep Time** \* 16 \* 1024 \* 1024 / Total Memory) for the interval between two scans, at least 10ms.

Reviewer 2: ```suggestion
- **Sleep Time(ms)**: The interval between two scans, which is calculated with the formula: **Sleep Time** \* 16 \* 1024 \* 1024 / Total Memory. Minimum: 10ms.
```

> **Assume you have a 512GiB memory node that uses the following parameters:**

Reviewer: ```suggestion
**For example, assume you have a 512GiB memory node that uses the following parameters:**
```

> **Here is a 512GiB memory node, assume the following parameters:**

Reviewer 2: ```suggestion
**Assume you have a 512GiB memory node that uses the following parameters:**
```

```yaml
Boost: 300
Decay: 100
Minimum Pages: 1000
Maximum Pages: 5000
Sleep Time: 50
```

When Ksmtuned starts, initialize `pages_to_scan` in KSM to 1000 (**Minimum Pages**) and set `sleep_millisecs` to 10 (50 \* 16 \* 1024 \* 1024 / 536870912 KiB < 10).

> KSM is started when the available memory is below the **Threshold Coefficient**, and if it is detected as running in each period (1 minute), `pages_to_scan` is incremented by 300 (**Boost**) until it reaches 5000 (**Maximum Pages**).

Reviewer: ```suggestion
KSM starts when free memory falls below the **Threshold Coefficient**. If it detects that it is running, `pages_to_scan` increments by 300 (**Boost**) every minute until it reaches 5000 (**Maximum Pages**).
```

> KSM is stopped when the available memory is higher than the **Threshold Coefficient**, and if it is detected as stopped in each period, `pages_to_scan` is decremented by 100 (**Decay**) until it reaches 1000 (**Minimum Pages**).

Reviewer: ```suggestion
KSM will stop when the free memory is above the **Threshold Coefficient**. If it detects that it is stopped, `pages_to_scan` decrements by 100 (**Decay**) every minute until it reaches 1000 (**Minimum Pages**).
```

> Start running the KSM when the available memory is below the **Threshold Coefficient**, and if it is detected as running every period (1 minute), `pages_to_scan` is incremented by 300 (**Boost**) until 5000 (**Maximum Pages**).
> 
> Stop running KSM when the available memory is higher than the **Threshold Coefficient**, and decrement `pages_to_scan` by 100 (**Decay**) until 1000 (**Minimum Pages**) if it detects that it has stopped each period.

Reviewer 2: ```suggestion
KSM is started when the available memory is below the **Threshold Coefficient**, and if it is detected as running in each period (1 minute), `pages_to_scan` is incremented by 300 (**Boost**) until it reaches 5000 (**Maximum Pages**).

KSM is stopped when the available memory is higher than the **Threshold Coefficient**, and if it is detected as stopped in each period, `pages_to_scan` is decremented by 100 (**Decay**) until it reaches 1000 (**Minimum Pages**).
```

Reviewer 2: We can use the same structure for these two sentences.
I changed these sentences according to my understanding. Make sure the meaning is still correct before you accept it.
